app/services/backup_code_generator.rb

```diff
@@ -12,16 +12,15 @@ def initialize(user, num_words: BackupCodeConfiguration::NUM_WORDS)
     @user = user
   end
 
-  # @return [Array<String>]
-  def generate
-    delete_existing_codes
-    generate_new_codes
-  end
+  def delete_and_regenerate(salt: SecureRandom.hex(32))
+    codes = generate_new_codes
+
+    BackupCodeConfiguration.transaction do
+      @user.backup_code_configurations.destroy_all
```

Comment:
`delete_all` would be slightly faster than `destroy_all` because it doesn't run callbacks for each object? and since backup codes are like leaf nodes, there shouldn't be anything to destroy?